feat: warn users when they relate an organisation to a non-active one

mirdono opened the pull request:
When a user tries to relate an organisation to another, non-active organisation
show a modal warning them about this. This modal requires the user to explicitly
confirm or cancel their choice.

## How to test
For existing organisations:
1. Launch an OP frontend with this branch
2. Log in as editor (either worship or non-worship)
4. Select an organisation
5. Navigate to the related organisations (*nl. Gerelateerde organisaties*) tab
   and click the edit button in the top right corner
6. Play around with adding relations with active and non-active organisations
   to see the modal appears when expected. Also try saving and cancelling
   changes to ensure no errors pop up in this flow.

For creating new organisations:
1. Launch an OP frontend with this branch
2. Log in as editor (either worship or non-worship)
3. On the organisations page click the "+ Nieuw" button in the top-right corner
   to create a new organisation.
4. Play around with creating different types of organisations. The relevant
   fields are located in the bottom part of the form in the "Gerelateerde
   organisaties" section. The exact contents will differ based on the selected
   "Type organisatie".

## Notes
- In this context "non-active" means any organisations with a status other than
  active (*nl. Actief*). Currently this covers organisations that are inactive
  (*nl. Niet Actief*) or in formation (*nl. In oprichting*).
- When the membership relations were introduced the required functionality was
  to only allow relations between active organisations. As part of this PR this
  rule is relaxed so users can select non-active organisations.

## Related tickets
- OP-3468

mirdono committed:
feat: allow relating an organisation to a non-active organisation

User can now select non-active organisations when adding relations between organisations.

mirdono committed:
feat: add confirmation when relating to a non-active organisation

When a user selects a non-active organisation as other entity in a membership,
show a modal information the user of this. The user has to explicitly confirm or cancel.

Windvis commented on app/components/related-organizations-edit-table.hbs:
```diff
@@ -110,6 +112,13 @@
           </AuButton>
         </td>
+
+        {{#if @confirmNonActiveRelatedOrganization}}
+          <ConfirmRelationWithInactiveOrganizationModal
+            @onConfirm={{fn @confirmNonActionRelatedOrganization membership}}
+            @onCancel={{fn @cancelNonActionRelatedOrganization membership}}
+          />
+        {{/if}}
```
Also, nitpick about the modal itself, but I think it might be better to not make the "Yes" button an alert? And maybe make the "nee" button a secondary or "naked" one, it feels wrong to have 2 primary buttons next to eachother. Feel free to ignore this if the current version was proposed by the design team :smile:.

![image](https://github.com/user-attachments/assets/ab698ca0-cecc-44a6-9b7d-b7fd688595e1)

Appuniversum has some docs about button usage it seems:
- https://appuniversum.github.io/ember-appuniversum/?path=/story/patterns-button-use--types
- https://appuniversum.github.io/ember-appuniversum/?path=/story/patterns-button-use--groups

mirdono replied:
> This seems to create a modal for each row in the table. It's hard to notice because they are all overlayed on top of each other (but the background grows darker with each modal that gets added).
> 
> I think it would be better to either move it to the route template (or alternatively move everything into the component instead, so this can be moved outside the table)?

Thanks for pointing this out. I moved the modal out the table component to avoid this.

mirdono replied:
> Also, nitpick about the modal itself, but I think it might be better to not make the "Yes" button an alert? And maybe make the "nee" button a secondary or "naked" one, it feels wrong to have 2 primary buttons next to eachother. Feel free to ignore this if the current version was proposed by the design team :smile:.
> 
> ![image](https://github.com/user-attachments/assets/ab698ca0-cecc-44a6-9b7d-b7fd688595e1)
> 
> Appuniversum has some docs about button usage it seems:
> - https://appuniversum.github.io/ember-appuniversum/?path=/story/patterns-button-use--types
> - https://appuniversum.github.io/ember-appuniversum/?path=/story/patterns-button-use--groups
> 
> That might be considered out of scope for this issue, but just mentioning here since I noticed it while testing.

Thanks for the pointers, I adjusted the buttons accordingly. (Another modal has the same issue but I will fix that in a separate PR.)

Windvis reviewed (changes requested):
Small remark about the implementation, but everything seems to work on the edit page.

The Jira issue also talks about the "new" organization flow (Acceptance criteria 1), but I don't see any code related to that in this PR. I don't know the codebase enough to know if that's still needed or not.

mirdono committed:
fix: allow relating new organisations to non-active ones

Business wants to be able to relate newly created organisations to non-active
organisations. This removes the restriction that previously prevented this.

mirdono committed:
chore: added TODO

mirdono committed:
feat: warn when creating an organisation related to a non-active one

When a user is creating a new organisation and relates that to a non-active
organisation show a warning modal that asks the user to confirm this.

For non-worship organisations, these relations concern the related municipality,
founders, and participants. The status of any related province is not explicitly
checked as there are no non-active provinces.

For worship organisations this concerns any related worship service, central
worship service, and representative body.

mirdono commented:
> Small remark about the implementation, but everything seems to work on the edit page.
> 
> The Jira issue also talks about the "new" organization flow (Acceptance criteria 1), but I don't see any code related to that in this PR. I don't know the codebase enough to know if that's still needed or not.

You are completely right, in my enthusiasm of putting this ticket into review I accidentally blew past he most complex part of this functionality. I added the confirmation modal in the new organisations form and updated the testing guidelines the PR's description.

Windvis commented on app/components/confirm-relation-with-inactive-organization-modal.hbs:
```diff
@@ -1,16 +1,21 @@
-        {{on "click" @onCancel}}
-      >Nee</AuButton>
-    </AuButtonGroup>
+    <AuToolbar as |Group|>
+      <Group />
```
Nitpick: AuToolbar has a `@reverse` argument that can be used to display the group on the right, which would have the same result without outputting unneeded html elements.

```suggestion
    <AuToolbar @reverse={{true}} as |Group|>
```

mirdono replied:
> Nitpick: AuToolbar has a `@reverse` argument that can be used to display the group on the right, which would have the same result without outputting unneeded html elements.
> 
> ```suggestion
>     <AuToolbar @reverse={{true}} as |Group|>
> ```

Thanks for the suggestion, that is indeed better. Not sure how I missed that option.

Windvis reviewed (approved):
Looks good :+1: 1 small nitpick, but it's not a blocker so feel free to merge as-is if you want.
